src/align/anchors.jl

```diff
@@ -0,0 +1,271 @@
+                    end
+                end
+            end
+        end
+
+        # compute first and last aligned reference positions
+        firstref = 0
+        for i in 1:length(anchors)
+            if ismatchop(anchors[i].op)
+                firstref = anchors[i-1].refpos
+            end
+        end
+
+        lastref = 0
+        for i in length(anchors):-1:1
+            if ismatchop(anchors[i].op)
+                lastref = anchors[i].refpos
+            end
+        end
+
+        return new(anchors, firstref, lastref)
+    end
+end
+
+
+"""
+Construct an `Alignment`
+"""
+function Alignment(cigar::AbstractString, seqpos::Int=1, refpos::Int=1)
+    # path starts prior to the first aligned position pair
+    seqpos -= 1
+    refpos -= 1
+
+    n = 0
+    anchors = AlignmentAnchor[AlignmentAnchor(seqpos, refpos, OP_START)]
+    for c in cigar
+        if isdigit(c)
+            n = n * 10 + convert(Int, c - '0')
+        else
+            if n == 0
+                error("CIGAR operations must be prefixed by a positive integer.")
+            end
+            op = Operation(c)
+            if ismatchop(op)
+                seqpos += n
+                refpos += n
+            elseif isinsertop(op)
+                seqpos += n
+            elseif isdeleteop(op)
+                refpos += n
+            else
+                error("The $(op) CIGAR operation is not yet supported.")
+            end
+
+            push!(anchors, AlignmentAnchor(seqpos, refpos, op))
+            n = 0
+        end
+    end
+
+    return Alignment(anchors)
+end
+
+
+function ==(a::Alignment, b::Alignment)
+    return a.anchors == b.anchors && a.firstref == b.firstref && a.lastref == b.lastref
+end
+
+
+function show(io::IO, aln::Alignment)
+    # print a representation of the reference sequence
+    anchors = aln.anchors
+    for i in 2:length(anchors)
+        if ismatchop(anchors[i].op)
+            for _ in anchors[i-1].refpos+1:anchors[i].refpos
+                write(io, '.')
+            end
+        elseif isinsertop(anchors[i].op)
+            for _ in anchors[i-1].seqpos+1:anchors[i].seqpos
+                write(io, '-')
+            end
+        elseif isdeleteop(anchors[i].op)
+            for _ in anchors[i-1].refpos+1:anchors[i].refpos
+                write(io, '.')
+            end
+        end
+    end
+    write(io, '\n')
+
+    # print a representation of the aligned sequence
+    for i in 2:length(anchors)
+        if ismatchop(anchors[i].op)
+            for i in anchors[i-1].seqpos+1:anchors[i].seqpos
+                write(io, '.')
+            end
+        elseif isinsertop(anchors[i].op)
+            for i in anchors[i-1].seqpos+1:anchors[i].seqpos
+                write(io, '.')
+            end
+        elseif isdeleteop(anchors[i].op)
+            for _ in anchors[i-1].refpos+1:anchors[i].refpos
+                write(io, '-')
+            end
+        end
+    end
+end
+
+
+immutable AlignedSequence{S <: Sequence}
```

Comment:
I think this `<: Sequence` restriction is too strict, we can align, for example, `ASCIIString`s.